google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/strategy/CreateNodesWithGroupEncodedIntoNameThenAddToSet.java

```diff
@@ -103,14 +103,15 @@
 
       String sharedResourceName = namingConvention.create().sharedNameForGroup(group);
       Template mutableTemplate = template.clone();
-      GoogleComputeEngineTemplateOptions templateOptions = GoogleComputeEngineTemplateOptions.class
-            .cast(mutableTemplate.getOptions());
+      GoogleComputeEngineTemplateOptions templateOptions = GoogleComputeEngineTemplateOptions.class.cast(mutableTemplate.getOptions());
       assert template.getOptions().equals(templateOptions) : "options didn't clone properly";
 
-      // get or insert the network and insert a firewall with the users
-      // configuration
+      // get or insert the network and insert a firewall with the users configuration
       Network network = getOrCreateNetwork(templateOptions, sharedResourceName);
-      getOrCreateFirewalls(templateOptions, network, firewallTagNamingConvention.get(group));
+      // add firewalls *only* if the template options don't contain explicit network
+      if (templateOptions.network() == null) {
```

Comment:
But it is not the same case. In OpenStack you have networks, and have security groups. By configuring the networks you are just saying to which networks a server should be attached. By configuring explicitly the security groups, you are telling exactly which "firewall policies" should apply to the server, so it makes sense to let that take precedence over the "inboundPorts".

Networks and security groups are not the same thing; one configures networking (ips, routing, etc), the other configures access (open ports, etc). It is not OK to say that attaching a server to a network implies that jclouds would ignore all access configuration to that server. In nova we don't do this; if the user explicitly configured a set of security groups (not related to networks), we use them to configure the access to the server, taking precedence over the inbound ports option, but the important thing here is that no one is mixing the network options and the firewall options.